tests/python/contrib/test_ethosu/infra.py

```diff
@@ -251,13 +251,23 @@ def verify_source(
     """
     interface_api = "c"
     test_runner = create_test_runner(accel, enable_usmp)
-    run_and_check(
-        models,
-        test_runner,
-        interface_api,
-        workspace_byte_alignment=16,
-        data_linkage=AOTDataLinkage(section="ethosu_scratch", alignment=16),
-    )
+
+    def run_mod():
+        run_and_check(
+            models,
+            test_runner,
+            interface_api,
+            workspace_byte_alignment=16,
+            data_linkage=AOTDataLinkage(section="ethosu_scratch", alignment=16),
+        )
+
+    # TODO(lhutton1) This is a quick and dirty work around to help temporarily reduce
+    # the flakyness of the tests. Will remove once #10300 is resolved.
+    try:
+        run_mod()
+    except AssertionError:
```

Comment:
@lhutton1 thanks for quickly taking this up!

We need to print out the logs if it fails (even once) if we are ever to fix this. i.e. we let it rerun but we do print the logs.
Generally, printing out run_log and compile_log would give the sufficient info to debug this.

Reply:
Or will it already happen by the time we land the other logging PR?

Reply:
I think we should try to land #10214 first, then rebase this patch and print the RuntimeError before the retry